macros/contexts/contextLinearRelation.pl

```diff
@@ -0,0 +1,399 @@
+displayed as C<<x+2 < y+z>> or converted to standard form: C<<x-y-z < -2>>.
+It is 0 by default.
+
+=back 
+
+There is one special method for LinearRelation objects.
+
+=over
+
+=item S<C<< $LR->check_at(point) >>>
+
+This returns true or false depending on if the point satisfies the relation.
+C<point> must be a MathObject Point, Vector, or ColumnVector; or simply be an
+array reference. The number of entries in C<point> must match the number of
+variables in the context.
+
+=back
+
+=cut
+
+loadMacros('MathObjects.pl');
+
+sub _contextLinearRelation_init { LinearRelation::Init() };    # don't reload this file
+
+##################################################
+#
+#  Initialize the contexts and make the creator function.
+#
+
+package LinearRelation;
+our @ISA = qw(Value::Formula);
+
+sub Init {
+	my $context = $main::context{LinearRelation} = Parser::Context->getCopy("Vector");
+	$context->{name}                       = "LinearRelation";
+	$context->{precedence}{LinearRelation} = $context->{precedence}{special};
+	$context->{value}{LinearRelation}      = "LinearRelation";
+	$context->{value}{Formula}             = "LinearRelation::formula";
+	$context->parens->remove("<");
+	$context->flags->set(standardForm => 0);
+	$context->operators->add(
+		'=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' = ',
+			kind          => 'eq',
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation"
+		},
+
+		'<' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' < ',
+			kind          => 'lt',
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation"
+		},
+		'>' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' > ',
+			kind          => 'gt',
+			reverse       => 'lt',
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation"
+		},
+
+		'<=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' <= ',
+			TeX           => '\le ',
+			kind          => "le",
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation",
+			alternatives  => ["\x{2264}"]
+		},
+		'=<' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' <= ',
+			TeX           => '\le ',
+			kind          => "le",
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation"
+		},
+
+		'>=' => {
+			precedence    => .5,
+			associativity => 'left',
+			type          => 'bin',
+			string        => ' >= ',
+			TeX           => '\ge ',
+			kind          => 'ge',
+			reverse       => 'le',
+			class         => 'LinearRelation::inequality',
+			formulaClass  => "LinearRelation",
+			alternatives  => ["\x{2265}"]
+		},
+		'=>' => {
```

Comment:
I'm not sure I like this.  I know this is an 'extra' way of doing '>=', but just looks like a perl hash.  Do any other language allow greater than or equal like this.   Also, same comment on `=<`.

Reply:
I doubt any student will say "Gee, that looks like a Perl hash" (or more likely, a Javascript function definition). It's not about programming languages, it's about what students are likely to type for the idea they are trying to get across. I agree that it is not pretty, but it is something students have typed in the past.

Reply:
Juts to make note here, this is so a student can enter an answer like `x + y ≥ 4`, either by pasting the `≥` character, by using a Mac keyboard `⌥ >`, or some other way of typing `≥` directly. And for that to be allowed, you would have to enable the "parseAlternatives" in localOverrides.conf.